internal/lookout/repository/jobs.go

```diff
@@ -262,27 +262,55 @@ func sortJobsByJobId(jobInfos []*lookout.JobInfo, descending bool) {
 	})
 }
 
-func determineJobStateDuration(state JobState, row *JobRow) string {
-	if row == nil {
-		return ""
-	}
+func (r *SQLJobRepository) updateTimeInState(jobInfo *lookout.JobInfo) {
 	var timeStamp time.Time
 
+	state := JobState(jobInfo.JobState)
 	switch state {
 	case JobSucceeded, JobFailed:
-		timeStamp = row.Finished.Time
+		possibleTime := findLatest(jobInfo.Runs, func(run *lookout.RunInfo) *time.Time { return run.Finished })
```

Comment:
Could you combine the if else loop into a common function?

```
		if possibleTime == nil {
			log.Warnf("No finished timestamp found for job with id %s", jobInfo.Job.Id)
		} else {
			timeStamp = *possibleTime
		}
```
The main difference between all these blocks of code is the error message which I think you can easily parametrize.